index.js

```diff
@@ -294,7 +297,18 @@ async function parsePipelineTemplate({ yaml }) {
 
     // flatten pipeline template shared setting into jobs
     pipelineTemplateConfig.jobs = flattenSharedIntoJobs(pipelineTemplateConfig.shared, pipelineTemplateConfig.jobs);
```

Comment:
at this point, actually shared steps have been merged into jobs regardless whether or not `screwdriver.cd/mergeSharedSteps` has been set to true, so in a later check of `handleMergeSharedStepsAnnotation`, even though it never go into [the if condition](https://github.com/screwdriver-cd/config-parser/blob/mergeSharedSteps/lib/phase/flatten.js#L322-L330), the shared steps will still be carried along.

Reply:
?